plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,613 @@
+        {
+        }
+
+        @Override
+        public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+        {
+            if (currentOnLoadTableCallCount + 1 == throwOnLoadTableCallCount) {
+                throw new TableNotFoundException(schemaTableName);
+            }
+
+            ++currentOnLoadTableCallCount;
+            return TEST_ICEBERG_TABLE;
+        }
+
+        @Override
+        public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+        {
+        }
+
+        @Override
+        public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+        {
+            return "";
+        }
+
+        @Override
+        public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTableName, TrinoPrincipal principal)
+        {
+        }
+
+        @Override
+        public void createView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorViewDefinition definition, boolean replace)
+        {
+        }
+
+        @Override
+        public void renameView(ConnectorSession session, SchemaTableName source, SchemaTableName target)
+        {
+        }
+
+        @Override
+        public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaViewName, TrinoPrincipal principal)
+        {
+        }
+
+        @Override
+        public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
+        {
+        }
+
+        @Override
+        public List<SchemaTableName> listViews(ConnectorSession session, Optional<String> namespace)
+        {
+            return ImmutableList.of();
+        }
+
+        @Override
+        public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> namespace)
+        {
+            return ImmutableMap.of();
+        }
+
+        @Override
+        public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewIdentifier)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        public List<SchemaTableName> listMaterializedViews(ConnectorSession session, Optional<String> namespace)
+        {
+            return ImmutableList.of();
+        }
+
+        @Override
+        public void createMaterializedView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+        {
+        }
+
+        @Override
+        public void dropMaterializedView(ConnectorSession session, SchemaTableName schemaViewName)
+        {
+        }
+
+        @Override
+        public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName schemaViewName)
+        {
+            return materializedViewDefinition;
+        }
+    }
+
+    private static class TestingTable
```

Comment:
nit: would it be possible to use real `Table implementation (e.g `org.apache.iceberg.BaseTable`)?

Reply:
I tried using `BaseTable`, it requires `HiveTableOperations` which would require initializing a `HiveMetastore`, and a few other dependencies.

  I think it might be simpler to use a mock table because we can specify what is returned easily.